tests/models/pegasus_x/test_modeling_pegasus_x.py

```diff
@@ -0,0 +1,852 @@
+        decoder_input_ids = _long_tensor([[0, 31414, 232, 328, 740, 1140, 12695, 69, 46078, 1588, 2]])
+        inputs_dict = prepare_pegasus_x_inputs_dict(model.config, input_ids, decoder_input_ids)
+        with torch.no_grad():
+            output = model(**inputs_dict)[0]
+        expected_shape = torch.Size((1, 11, model.config.vocab_size))
+        self.assertEqual(output.shape, expected_shape)
+        # change to expected output here
+        expected_slice = torch.tensor(
+            [[0.7144, 0.8143, -1.2813], [0.7144, 0.8143, -1.2813], [-0.0467, 2.5911, -2.1845]], device=torch_device
+        )
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=TOLERANCE))
+
+    def test_seq_to_seq_generation(self):
+        hf = PegasusXForConditionalGeneration.from_pretrained("zphang/pegasus-x-base").to(torch_device)
+        tok = PegasusTokenizer.from_pretrained("google/pegasus-large")
+
+        batch_input = [
+            "While large pretrained Transformer models have proven highly capable at tackling natural language tasks,"
+            " handling long sequence inputs continues to be a significant challenge. One such task is long input"
+            " summarization, where inputs are longer than the maximum input context of most pretrained models. Through"
+            " an extensive set of experiments, we investigate what model architectural changes and pretraining"
+            " paradigms can most efficiently adapt a pretrained Transformer for long input summarization. We find that"
+            " a staggered, block-local Transformer with global encoder tokens strikes a good balance of performance"
+            " and efficiency, and that an additional pretraining phase on long sequences meaningfully improves"
+            " downstream summarization performance. Based on our findings, we introduce PEGASUS-X, an extension of the"
+            " PEGASUS model with additional long input pretraining to handle inputs of up to 16K tokens. PEGASUS-X"
+            " achieves strong performance on long input summarization tasks comparable with much larger models while"
+            " adding few additional parameters and not requiring model parallelism to train."
+        ]
+
+        # The below article tests that we don't add any hypotheses outside of the top n_beams
+        dct = tok.batch_encode_plus(
+            batch_input,
+            max_length=512,
+            padding="max_length",
+            truncation_strategy="only_first",
+            truncation=True,
+            return_tensors="pt",
+        )
+
+        hypotheses_batch = hf.generate(
+            input_ids=dct["input_ids"].to(torch_device),
+            attention_mask=dct["attention_mask"].to(torch_device),
+            num_beams=2,
+            max_length=32,
+        )
+
+        EXPECTED = [
+            "we investigate the performance of a new pretrained model for long input summarization. <n> the model"
```

Comment:
Shouldn't there be more expected outputs ?  I don't really see how this test can pass at the moment

Reply:
It's a single long string (one example).

Reply:
Ah I see - good for me then! Guess it would have been nice to rename `batch_input` to just input since it's just one example but not a big deal!